src/components/RedirectionNotificationModal.js

```diff
@@ -1,72 +1,160 @@
 import Modal from "@mui/material/Modal";
+import { Link } from "react-router-dom/cjs/react-router-dom";
 import CloseRoundedIcon from "@mui/icons-material/CloseRounded";
 import * as s from "../components/RedirectionNotificationModal.sc";
+import { isMobile } from "../utils/misc/browserDetection";
 
 //This modal is used in the ArticlePreview component
 
 //TODO: Further refactor, e.g after testing and making sure users
 //understand text and wording, turn strings into variables and move to definitions.js
 
 export default function RedirectionNotificationModal({
+  api,
   article,
   open,
-  handleClose, //handleClose function defined in the ArticlePreview.js, passed as a prop
+  handleClose,
+  selectedDoNotShowRedirectionModal, //related to the "Do not show" checkbox selection
+  setSelectedDoNotShowRedirectionModal, //related to the "Do not show" checkbox selection
+  setOpenedExternallyWithoutModal, //related to the modal use based on the "Do not show" selection
+  setIsArticleSaved, // related to the article's state
 }) {
+  function handleVisibilityCheckboxSelection() {
+    setSelectedDoNotShowRedirectionModal(!selectedDoNotShowRedirectionModal);
+  }
+
+  //saves modal visibility preferences to the Local Storage
+  function handleModalUse() {
+    selectedDoNotShowRedirectionModal === true
+      ? setOpenedExternallyWithoutModal(true)
+      : setOpenedExternallyWithoutModal(false);
+  }
+
+  function handleCloseAndSavePreferences() {
+    handleModalUse();
+    handleClose();
+  }
+
+  function handleSaveArticle() {
+    api.makePersonalCopy(article.id, (data) => {
+      if (data === "OK") {
+        setIsArticleSaved(true);
+      }
+    });
+  }
+
+  function handleSaveAndOpenArticle() {
+    handleSaveArticle();
+    // handleModalUse(); //Temporarily disabled for this function on mobile as it worked only when <Link> had its target set to _blank
+    handleClose();
+  }
   return (
     <Modal open={open} onClose={handleClose}>
       <s.ModalWrapper>
-        <s.Header>
-          <h1>
-            You are ready to&nbsp;continue<br></br>
-            to the original article's website
-          </h1>
-        </s.Header>
-        <s.Body>
-          <p>
-            <strong>Once there</strong>, find and{" "}
-            <strong>
-              click The Zeeguu Reader{" "}
-              <s.Icon>
-                <img
-                  className="fullDivWidthImage"
-                  alt="The Zeeguu Reader elephant icon"
-                  src="../static/images/zeeguuLogo.svg"
-                ></img>
-              </s.Icon>{" "}
-              icon
-            </strong>{" "}
-            in the top right corner of&nbsp;your browser's toolbar
-            or&nbsp;on&nbsp;the&nbsp;list of your installed extensions{" "}
-            <s.Icon>
+        {!isMobile() ? (
```

Comment:
Wouldn't the code be more readable if we had a separate component here? Actually two components: `RedirectionNotificationForMobile` and `RedirectionNotifiactionForDesktop`?

Reply:
Fully agree. It's also one of my todos from the 05.09 update. :)